libdrgn/program.c

```diff
@@ -99,7 +99,10 @@ void drgn_program_deinit(struct drgn_program *prog)
 		else
 			drgn_thread_set_deinit(&prog->thread_set);
 	}
-	drgn_thread_destroy(prog->crashed_thread);
+	// When the target is a userspace core dump, `crashed_thread` will have been
+	// included in `prog->thread_set`, and thus freed by the above call to `drgn_thread_set_deinit`
+	if (!(prog->core_dump_notes_cached && !(prog->flags & DRGN_PROGRAM_IS_LINUX_KERNEL)))
```

Comment:
What is the `prog->core_dump_notes_cached` check for? Is it not sufficient to check `if (prog->flags & DRGN_PROGRAM_IS_LINUX_KERNEL)`?

Reply:
I forgot that `drgn_thread_destroy` actually will behave just fine when given `NULL`, will change.